sharpd/sharp_vty.c

```diff
@@ -81,46 +82,63 @@ DEFPY(watch_nexthop_v4, watch_nexthop_v4_cmd,
 
 DEFPY (install_routes,
        install_routes_cmd,
-       "sharp install routes A.B.C.D$start nexthop <A.B.C.D$nexthop4|X:X::X:X$nexthop6> (1-1000000)$routes [instance (0-255)$instance]",
+       "sharp install routes A.B.C.D$start [nexthop <A.B.C.D$nexthop4|X:X::X:X$nexthop6>|nexthop-group NAME$nexthop_group] (1-1000000)$routes [instance (0-255)$instance]",
        "Sharp routing Protocol\n"
        "install some routes\n"
        "Routes to install\n"
        "Address to start /32 generation at\n"
        "Nexthop to use(Can be an IPv4 or IPv6 address)\n"
        "V4 Nexthop address to use\n"
        "V6 Nexthop address to use\n"
+       "Nexthop-Group to use\n"
+       "The Name of the nexthop-group\n"
        "How many to create\n"
        "Instance to use\n"
        "Instance\n")
 {
 	int i;
 	struct prefix p;
 	struct nexthop nhop;
+	struct nexthop_group nhg;
 	uint32_t temp;
+	struct nexthop_group_cmd *nhgc;
 
 	total_routes = routes;
 	installed_routes = 0;
 
 	memset(&p, 0, sizeof(p));
 	memset(&nhop, 0, sizeof(nhop));
+	memset(&nhg, 0, sizeof(nhg));
 
 	p.family = AF_INET;
 	p.prefixlen = 32;
 	p.u.prefix4 = start;
 
-	if (nexthop4.s_addr != INADDR_ANY) {
-		nhop.gate.ipv4 = nexthop4;
-		nhop.type = NEXTHOP_TYPE_IPV4;
+	if (nexthop_group) {
+		nhgc = nhgc_find(nexthop_group);
```

Comment:
nit: you could reduce the scope of `nhgc` by declaring it in this block.

Reply:
fixed